### DISCUSSION TITLE
SOLVED! Philips Hue Wall Switch Module won't pair!

### DISCUSSION THREAD

**Discussion by GJayBoi** [Koenkk/zigbee2mqtt#16706] [Devices]
Hi,
So far I have wasted too much time on this so I thought I would ask...
I have a [Phillips Wall Switch Module](https://www.zigbee2mqtt.io/devices/929003017102.html), and I am trying to add it to 2MQTT (In home assistant) I have other Hue devices, and various other manufacturers, and some none lighting devices (TRV's), I can join other devices happily but this wall module just will not, I put it into pairing mode... I allow new devices to join in 2MQTT, and nothing, it doesn't even try to connect.
I can re-add it to the Phillips Hue Hub, and set it up, but I don't want it on that device any longer, I now have a mix of lighting and I want to deprecate the Hue Hub soon, so will move everything over, so far everything else including a "Remote Dimmer" has all worked, it's just this one darn thing!
I have the SONOFF 3.0 +  Dongle directly in HA.
Any suggestions would be much appreciated, it's been doing my head in 😠 SOOO. for now I am putting it down and leaving a hole in the wall where the switch should be 😄 
Ta,
JC.

**Comment by donutVillan** [comment]
I'm using several of these with Z2M and HA. Never had a problem pairing them.
I've done it both with the Sonoff ZBDongle-E and the Sonoff ZBDongle-P. I did however have a problem with them "leaving" the network when using the Sonoff ZBDongle-E and re-starting HA. That's why I changed to the P version.
I don't know how you pair them with the Hue bridge but with Z2M you have to toggle the left button. The right one does not work. And you have to toggle it several times.

**Reply by GJayBoi** [reply]
Hey @donutVillan 
Thanks for the response, yeah paring with Hue is easy it goes straight back on after a reset, and I literally flip the switch once, but with Z2M it just doesn't wanna do it, but I have to admit I haven't toggled it several times!!!
Could I just confirm, you say the left button, I am using mine on a single switch, so it's plugged into "input 1" on the Switch Module, my brain says this would be the left switch (it's certainly where I would put it) but just to be through I may well plug the little leads into input 2 and just short (switch / press) them as well and see if that makes any difference!
Good to know that someone has proven it to work at least, if I can't it suggests that either my module is not behaving as expected or something is definitely wrong this end.

**Reply by donutVillan** [reply]
If I remember correctly then left is input 1 as you say. I can check tonight when I’m back home again.
I definitely had to press the button several times to pair it. Maybe it would have paired anyway over time with just one press but I didn’t want to wait for that. 5-10 times over the course of a few seconds worked for me.

**Comment by GJayBoi** [comment]
I have it working!! 😄 
So I tried as I mentioned above, and fiddling with "input 2" and nothing, I also tried toggling "input 1" several times, and despite the device being in pair mode (the light was flashing) there was nothing.....

Then... I spent some time digging through the Settings for Z2M and under MQTT at the bottom is this little check box called "Reject unauthorised" I had seen this yesterday but ignored it assuming that when you "permit join all" this isn't relevant, but seems it is... when I uncheck it and save the config, the device joined immediately!

![Screenshot 2023-02-17 at 12 58 52](https://user-images.githubusercontent.com/6206501/219660073-9a27abf4-24c6-4a92-b218-0a861653de15.png)
